### PR TITLE
[APPAI-1447] Part 1 :: User validation based on UUID with decorator

### PR DESCRIPTION
**Pull request by dgpatelgit**
Handle UUID in request header for registered snyk user making stack analyses request.

Jira issue :: https://issues.redhat.com/browse/APPAI-1447

### CODE REVIEW

File: bayesian/auth.py [hunk 1]
```diff
@@ -26,3 +29,40 @@ def get_access_token(service_name):
 
     except Exception:
         logger.error('Unable to connect to Auth service')
+
+
+def get_user_type(view):
```
Comment (arajkumar) [inline bayesian/auth.py:34]: I guess this should be named as `validate_user`

File: bayesian/auth.py [hunk 2]
```diff
@@ -26,3 +29,40 @@ def get_access_token(service_name):
...
+        if uuid is not None:
+            try:
+                # Read user details from RDS based on uuid
+                user = get_user(uuid)
+                if user.status == 'REGISTERED':
+                    g.registration_status = RegistrationStatus.registered
+            except Exception as e:
+                logger.warning('User get for UUID: %s failed with %s', uuid, e)
+
+        logger.debug('For UUID: %s, got user type: %s', uuid, g.registration_status)
```
Comment (arajkumar) [inline bayesian/auth.py:65]: Try block should have a block which would really generate an Exception.

```suggestion
if uuid:
    try:
        user = get_user(uuid)
    except <ProperExceptionClass_NOT_GenericException> e:
        logger.warning("Unable to get status for uuid=%s, err=%s", uuid, e)
    else:
        g.registration_status = RegistrationStatus[user.status]
     logger.debug('For UUID: %s, got user type: %s', uuid, g.registration_status)
```

File: bayesian/utility/v2/sa_models.py [hunk 3]
```diff
@@ -31,6 +31,13 @@ class Ecosystem(str, Enum):
     npm = 'npm'
 
 
+class RegistrationStatus(str, Enum):
```
Comment (prashbnair) [inline bayesian/utility/v2/sa_models.py:34]: Can you use the ENUM in user_utils

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: bayesian/utility/v2/sa_models.py (hunk 3 above)
Comment (dgpatelgit) [reply bayesian/utility/v2/sa_models.py:34]: There is difference between status value used in Backbone flow and user utils, we need to have mapping somewhere. We can discuss on this in standup and close it.

**Comment by dgpatelgit** [comment]
> There are two possible solution to generate different response based on user registration status
> 
> 1. Backbone generates response according to registration status
> 2. Backbone can always generate registered user response but API server filters it based on registration status
> 
> Current backbone implementation is based on #1, however #2 seems to be the optimal solution because we can easily switch from unregistered response to registered response without re triggering the full stack analysis.

@arajkumar  As discussed in meeting, we will implement approach #2 and filtering of the report shall be done by API server. I will raise a separate PR with this changes, meanwhile, CA changes needs common implementation of decorator function present in this PR, so I have modified this PR with only user validation decorator changes. PTAL.

File: bayesian/auth.py [hunk 4]
```diff
@@ -26,3 +28,41 @@ def get_access_token(service_name):
...
+        # Check user with uuid if it is present in RDS.
+        if uuid is not None:
+            try:
+                # Read user details from RDS based on uuid
+                user = get_user(uuid)
+            except UserException as e:
+                logger.warning("Unable to get status for uuid=%s, err=%s", uuid, e)
+            else:
+                if user.status == 'REGISTERED':
+                    g.user_status = UserStatus.REGISTERED
```
Comment (arajkumar) [inline bayesian/auth.py:63]: Why do we have two formats (types) to represent User status? one is string 'REGISTERED/UNREGISTERED' and another is enum. This should be unified.

File: bayesian/auth.py [hunk 5]
```diff
@@ -26,3 +28,41 @@ def get_access_token(service_name):
...
+        # By default set this to 'freetier'.
+        g.registration_status = UserStatus.FREETIER
+
+        # Read uuid from request header.
+        uuid = request.headers.get('uuid', None)
+
+        # Check user with uuid if it is present in RDS.
+        if uuid is not None:
+            try:
+                # Read user details from RDS based on uuid
```
Comment (arajkumar) [inline bayesian/auth.py:57]: I don't recommend to comment on each and every operation unless it is not expressible using code. I feel the code is good enough and comments are redundant.

File: bayesian/auth.py (hunk 4 above)
Comment (arajkumar) [inline bayesian/auth.py:63]: When dealing with enum, you can use a simple approach to map to values.

```suggestion
                g.user_status = UserStatus[user.status]
```

**Review by arajkumar — CHANGES_REQUESTED** [review]

**Pull request merged**